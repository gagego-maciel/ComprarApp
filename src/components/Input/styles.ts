import styled from 'styled-components/native'

export const Container = styled.View`
  width: 100%;
`

export const InputLabel = styled.Text`
  font-size: 14px;
  font-weight: 600;
  color: ${({ theme }) => theme.colors.textBlack};
  margin-bottom: 4px;
`

export const Field = styled.TextInput.attrs((prop) => ({
  placeholderTextColor: prop.theme.colors.placeHolderInput,
}))`
  width: 100%;
  height: 48px;
  padding: 0 16px;
  border-width: 1px;
  border-color: ${({ theme }) => theme.colors.border};
  border-radius: 8px;
  font-size: 16px;
  font-weight: 500;
  color: ${({ theme }) => theme.colors.textBlack};
  background-color: ${({ theme }) => theme.colors.backgroundWhite};
`
