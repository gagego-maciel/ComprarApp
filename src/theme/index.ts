export const theme = {
  colors: {
    primary: '#8257E5',
    background: '#121214',
    backgroundWhite: '#ffffff',
    backgroundSuccess: '#27ae60',
    backgroundButtonBlue: '#2c46b1',
    backgroundButtonRed: '#E53E3E',
    text: '#E1E1E6',
    textBlack: '#000000',
    textWhite: '#ffffff',
    textGray: '#A0A0A0',

    border: '#c3c5cb',

    placeHolderInput: '#74798B',
  },
}
