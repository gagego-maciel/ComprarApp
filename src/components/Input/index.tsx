import * as S from './styles'

interface IInputProps {
  label?: string
  placeholderText?: string
  value: string
  onChangeText: (value: string) => void
  maxLength?: number
}

export const Input: React.FC<IInputProps> = ({
  label,
  placeholderText,
  value,
  onChangeText,
  maxLength,
}) => {
  return (
    <S.Container>
      {label && <S.InputLabel>{label}</S.InputLabel>}

      <S.InputField
        placeholder={placeholderText}
        value={value}
        onChangeText={onChangeText}
        maxLength={maxLength}
      />
    </S.Container>
  )
}
