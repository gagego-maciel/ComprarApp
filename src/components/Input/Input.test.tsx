import React from 'react'
import { theme } from '@/theme'
import { Input, type IInputProps } from '../Input'
import { ThemeProvider } from 'styled-components/native'
import { render, fireEvent } from '@testing-library/react-native'

const renderWithTheme = (props: IInputProps) =>
  render(
    <ThemeProvider theme={theme}>
      <Input {...props} />
    </ThemeProvider>,
  )

describe('<Input />', () => {
  const mockOnChangeText = jest.fn()

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should render without crashing', () => {
    const { getByPlaceholderText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
      placeholder: '',
    })
    expect(getByPlaceholderText('')).toBeTruthy()
  })

  it('should render the label when provided', () => {
    const labelText = 'Username'
    const { getByText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
      label: labelText,
      placeholder: '',
    })
    expect(getByText(labelText)).toBeTruthy()
  })

  it('should not render the label when not provided', () => {
    const { queryByText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
    })
    expect(queryByText(/./)).toBeNull()
  })

  it('should display the correct placeholder', () => {
    const placeholder = 'Enter text'
    const { getByPlaceholderText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
      placeholder,
    })
    expect(getByPlaceholderText(placeholder)).toBeTruthy()
  })

  it('should display the correct value', () => {
    const value = 'Initial value'
    const { getByDisplayValue } = renderWithTheme({
      value,
      onChangeText: mockOnChangeText,
    })
    expect(getByDisplayValue(value)).toBeTruthy()
  })

  it('should call onChangeText when typing', () => {
    const placeholder = 'Type here'
    const { getByPlaceholderText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
      placeholder,
    })

    const input = getByPlaceholderText(placeholder)
    fireEvent.changeText(input, 'Hello')

    expect(mockOnChangeText).toHaveBeenCalledWith('Hello')
  })

  it('should respect maxLength prop', () => {
    const placeholder = 'Max 5'
    const { getByPlaceholderText } = renderWithTheme({
      value: '',
      onChangeText: mockOnChangeText,
      placeholder,
      maxLength: 5,
    })

    const input = getByPlaceholderText(placeholder)
    fireEvent.changeText(input, '12345')

    expect(mockOnChangeText).toHaveBeenCalledWith('12345')
  })

  it('should match snapshot', () => {
    const { toJSON } = renderWithTheme({
      value: 'Snapshot',
      onChangeText: mockOnChangeText,
      label: 'Snapshot Label',
      placeholder: 'Snapshot Placeholder',
    })

    expect(toJSON()).toMatchSnapshot()
  })
})
