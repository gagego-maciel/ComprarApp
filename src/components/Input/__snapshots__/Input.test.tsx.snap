// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should match snapshot 1`] = `
<View
  style={
    {
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 14,
        "fontWeight": "600",
        "marginBottom": 4,
      }
    }
  >
    Snapshot Label
  </Text>
  <TextInput
    accessibilityLabel="Snapshot Label"
    onChangeText={[MockFunction]}
    placeholder="Snapshot Placeholder"
    placeholderTextColor="#74798B"
    style={
      {
        "backgroundColor": "#ffffff",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#c3c5cb",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 1,
        "color": "#000000",
        "fontSize": 16,
        "fontWeight": "500",
        "height": 48,
        "paddingBottom": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 0,
        "width": "100%",
      }
    }
    testID="Snapshot Label"
    value="Snapshot"
  />
</View>
`;
